src/ledger/NetworkConfig.h

```diff
@@ -175,9 +179,32 @@ class SorobanNetworkConfig
     // General execution ledger settings
     uint32_t ledgerMaxTxCount() const;
 
+    // Number of samples in slidign window
+    uint32_t getBucketListSizeSnapshotPeriod() const;
+
+    // If newSize is different than the current BucketList size sliding window,
+    // update the window. If newSize < currSize, pop entries off window. If
+    // newSize > currSize, add as many copies of the current BucketList size to
+    // window until it has newSize entries.
+    void maybeUpdateBucketListWindowSize(AbstractLedgerTxn& ltx,
```

Comment:
I think this should be a static/free function and not fields should be mutable. Instead, this would just update the ledger entry and then we will automatically update the fields as a part of `SorobanNetworkConfig::loadFromLedger`  (it happens after any update).

I think it's not necessary to do the change in this PR though - probably just open an issue to do cleanup?